stock_picking_invoice_link/models/account_invoice.py

```diff
@@ -19,16 +19,28 @@ class AccountInvoice(models.Model):
              "(only when the invoice has been generated from a sale order).",
     )
 
+    @api.model
+    def _refund_cleanup_lines(self, lines):
+        result = super(AccountInvoice, self)._refund_cleanup_lines(lines)
+        if self.env.context.get('mode') == 'modify':
+            for i, line in enumerate(lines):
```

Comment:
Isn't there any `_prepare_refund_line` or similar for not using this weak link through line index?

Reply:
I think that this ok